src/app/api/fruit_locations/route.ts

```diff
@@ -237,17 +248,17 @@ export async function GET(request: Request) {
 export async function POST(request: Request) {
     const data = await request.json();
 
-    if(!data || !data.name || !data.latitude || !data.longitude || !data.fruit_id) {
+    if(!data || !data.name || !data.latitude || !data.longitude || !data.fruit_id || !data.user_id) {
```

Comment:
It would be more secure to get the `user_id` from the user's server side session.  Allowing it to be passed in lets people impersonate others.